ledger/verify_block.go

```diff
@@ -23,124 +23,335 @@ import (
 	"encoding/hex"
 	"errors"
 	"fmt"
+	"math"
+	"os"
 
 	"github.com/blinklabs-io/gouroboros/cbor"
+	"github.com/blinklabs-io/gouroboros/ledger/allegra"
+	"github.com/blinklabs-io/gouroboros/ledger/alonzo"
+	"github.com/blinklabs-io/gouroboros/ledger/babbage"
+	"github.com/blinklabs-io/gouroboros/ledger/byron"
+	"github.com/blinklabs-io/gouroboros/ledger/common"
+	"github.com/blinklabs-io/gouroboros/ledger/conway"
+	"github.com/blinklabs-io/gouroboros/ledger/mary"
+	"github.com/blinklabs-io/gouroboros/ledger/shelley"
+	"golang.org/x/crypto/blake2b"
 )
 
-//nolint:staticcheck
-func VerifyBlock(block BlockHexCbor) (bool, string, uint64, uint64, error) {
-	headerCborHex := block.HeaderCbor
-	epochNonceHex := block.Eta0
-	bodyHex := block.BlockBodyCbor
-	slotPerKesPeriod := uint64(block.Spk) // #nosec G115
+const (
+	HeaderBodyLengthShelleyLike = 15
+	HeaderBodyLengthBabbageLike = 10
+	ProtoMajorShelley           = 2
+	ProtoMajorAllegra           = 3
+	ProtoMajorMary              = 4
+	ProtoMajorAlonzo            = 5
+	ProtoMajorBabbage           = 7
+	ProtoMajorConway            = 9
+)
+
+var allowMissingCbor = false
 
+// DetermineBlockType determines the block type from the header CBOR
+func DetermineBlockType(headerCbor []byte) (uint, error) {
```

Comment:
This function was only used by the now-removed `parseHeader()`. The constants above are (probably) only used by this function and can also be removed.

Reply:
In ledger/verify_block_test.go line 113:
```
blockType, err := DetermineBlockType(headerBytes)
```